app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -314,12 +314,36 @@ class FormController(
         form = cache.form.copy(visitsIndex = VisitIndex(visitsIndex), componentIdToFileId = componentIdToFileId)
       )
 
-      validateAndUpdateData(cacheUpd, processDataUpd, sn, sn, maybeAccessCode, ff, formModelOptics) { _ =>
-        val sectionTitle4Ga = getSectionTitle4Ga(processDataUpd, sn)
-        Redirect(
-          routes.FormController
-            .form(cache.formTemplate._id, maybeAccessCode, sn, sectionTitle4Ga, SuppressErrors.Yes, FastForward.Yes)
-        )
+      validateAndUpdateData(cacheUpd, processDataUpd, sn, sn, maybeAccessCode, ff, formModelOptics) {
+        maybeSectionNumber =>
+          val sectionTitle4Ga = getSectionTitle4Ga(processDataUpd, sn)
+
+          val sectionNumber =
```

Comment:
It is hard to describe this, so here is code snippet with small refactoring:
```
val addToListBracket: Bracket.AddToList[DataExpanded] = updFormModel.brackets.addToListBracket(addToListId)

val sectionNumber =
  if (addToListBracket.iterations.size === 1)
    maybeSectionNumber.map(addToListBracket.iterationForSectionNumber(_).firstSectionNumber).getOrElse(sn)
  else
    sn 
```

Notice `iterations` instead of `repeaters` as well.